server/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalNumericMetricsAggregation.java

```diff
@@ -79,7 +79,12 @@ protected MultiValue(StreamInput in) throws IOException {
         public abstract double value(String name);
 
         public String valueAsString(String name) {
-            return format.format(value(name)).toString();
+            // Explicitly check for NaN, since it formats to "�" or "NaN" depending on JDK version
```

Comment:
Is this only the case for certain locales? I would bne suprised if some JDKs would return a weird UTF8 character in all cases. To make this comment more readable it would probably also make sense to put in the bad utf8 value as octal or hex codepoint and to clarify under which circumstances this happens.

Reply:
I'm not actually sure how this behaves across Locales, but I don't think it matters for us.  We seem to always initialize the Decimal `DocValueFormat` [with `Locale.Root`](https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/search/DocValueFormat.java#L366) which I believe uses the JRE's default symbol table.  

So for JDK8 the root locale will use `JRELocaleProviderAdapter` to get the symbols, which loads `sun.text.resources.FormatData`, and you can see the [`NaN` symbol is `\uFFFD`](http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/sun/text/resources/FormatData.java#l269)

For JDK 9+, the root locale will use `CLDRLocaleProviderAdapter`, which loads `sun.text.resources.cldr.FormatData`.  And in that resource file you can see the `NaN` symbol is `"NaN"` (Can't find a link to the code, but you can see it in your IDE).

++ to making the comment more descriptive.  I'll try to distill this thread into a sane comment, and probably leave a reference to the comments here in case anyone wants to see more info.

Reply:
As an aside, I really wonder why Oracle thought � would be a good default representation of "NaN"... :(

Reply:
Hm, I don't like how this was implemented, looking at it.  Going to move it over to the DocValueFormat itself, so that it only applies to the Decimal formatter when looking at doubles... otherwise it'll be checked against all formatters (geo, IP, etc).  Harmless I think, but no need.